r2dbc-spi/src/main/java/io/r2dbc/spi/R2dbcBadGrammarException.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.spi;
+
+/**
+ * Exception thrown when the SQL statement has a problem in its syntax.
+ */
+public class R2dbcBadGrammarException extends R2dbcNonTransientException {
```

Comment:
We should add the offending SQL statement into this exception. This would improve debugging because we can directly correlate which statement caused the issue. Probably not ideal when using combined statements (`SELECT TOP FROM foobar; SELECT * FROM baz`), but still there is a lot of value.

Reply:
Hello @mp911de 

I was wondering the best way of doing this. Maybe adding another constructor? Should this new constructor append the current SQL to the error exception?

Reply:
I think we should keep the SQL separate from the message. Let’s add two constructors that take all args and the SQL and the second constructor would additionally take a cause. Typically, we would have a SQL state, an error, the message and the offending SQL.